src/Commands/Connection/InfoCommand.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Pantheon\Terminus\Commands\Connection;
+
+use Consolidation\OutputFormatters\StructuredData\RowsOfFields;
+
+use Pantheon\Terminus\Commands\TerminusCommand;
+use Terminus\Collections\Sites;
+use Terminus\Models\Environment;
+
+/**
+ * Class InfoCommand
+ * @package Pantheon\Terminus\Commands\Connection
+ */
+class InfoCommand extends TerminusCommand
+{
+    /**
+     * Retrieve connection info for a specific environment such as git, sftp, mysql, redis
+     *
+     * @authorized
+     *
+     * @command connection:info
+     *
+     * @param string $environment Name of the environment to retrieve
+     * @param string $filter Parameter filter (optional)
+     * @param array $options [fields=<env,param,value>] [format=<table|csv|yaml|json>]
+     *
+     * @return RowsOfFields
+     *
+     * @field-labels
+     *   env: Environment
+     *   param: Parameter
+     *   value: Connection Info
+     *
+     * @example connection:info awesome-site dev git_command --format=json
+     *   Display connection information only for the given parameter
+     *
+     */
+    public function connectionInfo(
+        $environment,
+        $filter = null,
+        $options = ['format' => 'table', 'fields' => 'param,value']
+    ) {
+        $site_env = explode('.', $environment);
```

Comment:
If the site and env are going to sometimes be optional for some commands, you might want to consider using the `@` prefix in Terminus, like Drush uses.  This allows Drush to easily determine if the site alias is present or not.  Of course, with Terminus, you have a site specification for most commands -- although some are site only, and others are site and env.